### PR TITLE
New widget: inline comments history

### PR DESCRIPTION
**Pull request by pablopunk**
SDFID-297

Okay this is how it looks:

captura de pantalla 2018-03-05 a las 13 34 32

### CODE REVIEW

File: scripts/apps/authoring/inline-comments/inline-comments.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    const users = {};
+
+    comments.forEach(({data}) => {
+        users[data.authorId] = null;
+        users[data.resolutionInfo.resolverUserId] = null;
+        data.replies.map((reply) => users[reply.authorId] = null);
+    });
+
+    return Object.keys(users);
+}
```
[the remark this answers is not in the document]
Comment (pablopunk) [reply scripts/apps/authoring/inline-comments/inline-comments.js]: That would be an array with a lot of duplicated user Ids. Also at first I did save a value (not null), but I could remove it for something like this:

```js
function getAllUserIdsFromCommentsAndReplies(comments) {
    let users = [];

    comments.forEach(({data}) => {
        if (!users.includes(data.authorId)) {
           users.push(data.authorId);
        }
        data.replies.foreach((reply) => {
            if (!users.includes(reply.authorId)) {
               users.push(reply.authorId);
            }
        });
    });

    return users;
}
```

But I'm not sure if it's any better

File: scripts/apps/authoring/inline-comments/inline-comments.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+function injectUsersInComments(users, comments) {
```
[the remark this answers is not in the document]
Comment (pablopunk) [reply scripts/apps/authoring/inline-comments/inline-comments.js]: I guess I could expose `$scope.users` to the view

File: scripts/apps/authoring/inline-comments/inline-comments.scss [hunk 3]
```diff
@@ -0,0 +1,56 @@
+.sd-widget.inline-comments {
+    .notification-list {
+        position: absolute;
```
[the remark this answers is not in the document]
Comment (pablopunk) [reply scripts/apps/authoring/inline-comments/inline-comments.scss]: Yeah all CSS was copied from the other comments widget, that's why I wanted to rethink the CSS structure so we don't repeat so much code in such similar things

File: scripts/apps/authoring/inline-comments/inline-comments.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (tomaskikutis) [reply scripts/apps/authoring/inline-comments/inline-comments.js]: Got it now. Also, I forgot the `resolverId` in my example. You could use my example and filter it later with lodash, check before adding like in the example you provided or keep your initial version, but with a comment explaining what you are doing. It seems clear now, but it didn't when I was first reviewing it.

File: scripts/apps/authoring/inline-comments/inline-comments.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (tomaskikutis) [reply scripts/apps/authoring/inline-comments/inline-comments.scss]: If that's the case, check if you can refactor it all to use the same styles in a reasonable time, and do that if you can. If it's multiple days of work, leave it as it is and we'll solve it when it gets in the way next time.

File: scripts/apps/authoring/inline-comments/views/inline-comments-widget.html [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+                        <img sd-user-avatar data-user="users[reply.authorId]">
+                    </figure>
+                    <div class="content">
+                        <div class="text" sd-comment-text data-comment="comment" data-name="{{users[reply.authorId].display_name || users[reply.authorId].username}}" data-text="{{reply.msg}}"></div>
+                        <span class="date" sd-reldate ng-model="reply.date"></span>
+                    </div>
+                </li>
+            </ul>
+            <div class="resolution">
+                <span>Resolved by {{ users[item.data.resolutionInfo.resolverUserId].display_name || users[item.data.resolutionInfo.resolverUserId].username }} <span class="date" sd-reldate ng-model="item.data.resolutionInfo.date"></span></span>
```
Comment (petrjasek) [inline scripts/apps/authoring/inline-comments/views/inline-comments-widget.html]: there is missing translate: `{{ :: 'Resolved by' | translate }} {{ user... `

File: scripts/apps/authoring/inline-comments/inline-comments.scss [hunk 5]
```diff
@@ -1,48 +1,11 @@
 .sd-widget.inline-comments {
     .notification-list {
-        position: absolute;
-        bottom: 110px;
         top: 0;
-        left: 0;
         right: 0;
+        left: 0;
```
Comment (tomaskikutis) [inline scripts/apps/authoring/inline-comments/inline-comments.scss]: If you removed `position:absolute` shouldn't `top` `left` and `right` also be removed?

Comment (pablopunk) [reply scripts/apps/authoring/inline-comments/inline-comments.scss]: You're right, but it seems `.notification-list` is always `position: absolute`. I could overwrite this or make my own `.comment-list` but I think it's better to reuse it for now and then refactor `.notification-list` if necessary.

**Pull request merged**